pkg/trtl/honu.go

```diff
@@ -111,8 +112,74 @@ func (h *HonuService) Iter(ctx context.Context, in *pb.IterRequest) (out *pb.Ite
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
 
-func (h *HonuService) Batch(stream pb.Trtl_BatchServer) (err error) {
-	return status.Error(codes.Unimplemented, "not implemented")
+// Batch is a client-side streaming request to issue multiple commands, usually Put and Delete.
+func (h *HonuService) Batch(stream pb.Trtl_BatchServer) error {
+	out := &pb.BatchReply{}
```

Comment:
There is a hidden place where errors might come from, and that's `stream.Context()` which returns a context object. That context has a `ctx.Done()` channel, which will inform you if the stream is closing, usually because of a deadline exceeded error. That error may not show up as a non-EOF error in `Recv`, so some experimentation is required to determine how we want to handle receiving messages and dealing with deadlines from the database. We can discuss this during our next sync.